src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -334,11 +335,15 @@ protected virtual TxReceipt[] ProcessBlock(
     [MethodImpl(MethodImplOptions.NoInlining)]
     private static void CalculateBlooms(TxReceipt[] receipts)
     {
-        int index = 0;
-        Parallel.For(0, receipts.Length, _ =>
+        ParallelUnbalancedWork.For(
+            0,
+            receipts.Length,
+            ParallelUnbalancedWork.DefaultOptions,
+            receipts,
+            static (i, receipts) =>
```

Comment:
If you would make a `ForEach` version, then you don't need to pass any state to it.
Or maybe by default (or by overload) also pass the element to the delegate.

Reply:
ForEach would need locking on the iterator so introduce contention?